### PR TITLE
Split layout

### PR DESCRIPTION
**Pull request by dokato**
Another step in creating layouts. Split layout that splits screen into `N` columns added.

Shiny version:

```r
  ui <- fluidPage(
    splitLayout(
      cellWidths = 300,
      cellArgs = "padding: 6px;",
      style = "border: 1px solid silver;",
      plotOutput("plot1"),
      plotOutput("plot2"),
      plotOutput("plot3")
    )
  )
```
Screenshot 2020-08-01 at 11 09 58

Semantic version:
```r
  ui <- semanticPage(
    splitLayout(
      cellWidths = 300,
      cellArgs = "padding: 6px;",
      style = "border: 1px solid silver;",
      plotOutput("plot1"),
      plotOutput("plot2"),
      plotOutput("plot3")
    )
  )
```

Screenshot 2020-08-01 at 11 11 26

**DoD**

- [x] Major project work has a corresponding task. If there’s no task for what you are doing, create it. Each task needs to be well defined and described.

- [x] Change has been tested (manually or with automated tests), everything runs correctly and works as expected. No existing functionality is broken.

- [x] No new error or warning messages are introduced.

- [x] All interaction with a semantic functions, examples and docs are written from the perspective of the person using or receiving it. They are understandable and helpful to this person.

- [x] If the change affects code or repo sctructure, README, documentation and code comments should be updated. 

- [x] All code has been peer-reviewed before merging into any main branch.

- [x] All changes have been merged into the main branch we use for development (develop).

- [x] Continuous integration checks (linter, unit tests) are configured and passed.

- [x] Unit tests added for all new or changed logic.

- [x] All task requirements satisfied. The reviewer is responsible to verify each aspect of the task.

- [x] Any added or touched code follows our style-guide.

### CODE REVIEW

File: R/layouts.R [hunk 1]
```diff
@@ -162,3 +162,96 @@ sidebarLayout <- function(sidebarPanel,
...
+split_layout <- function(..., cell_widths = NULL, cell_args = "", style = NULL){
+  if (class(cell_args) == "list")
+    stop("In this implementation of `split_layout` cell_args must be character with style css")
+  ui_elements <- list(...)
+  n_elems <- length(ui_elements)
+  columns <- paste0("col", seq(1, n_elems))
+  names(ui_elements) <- columns
+  if (is.null(cell_widths)) {
+    split_width <- floor(100/n_elems)
+    cell_widths <- mapply(function(x) paste0(x, "%"), rep(split_width, n_elems))
```
Comment (przytu1) [inline R/layouts.R:233]: I am not sure why `mapply` is used here. Shouldn't just this work:

```
cell_widths <- paste0(rep(split_width, n_elems), "%")
```

File: R/layouts.R [hunk 2]
```diff
@@ -162,3 +162,96 @@ sidebarLayout <- function(sidebarPanel,
...
+  }
+  layout <- grid_template(default = list(
+    areas = rbind(columns),
+    cols_width = cell_widths
+    )
+  )
+  if (is.null(style))
```
Comment (przytu1) [inline R/layouts.R:240]: How about `ifelse` version?

```
container_style <- ifelse(is.null(style), "background: #d8d8d8; margin: 5px;", style)
```

or

```
container_style <- if (is.null(style)) "background: #d8d8d8; margin: 5px;" else style
```

File: R/layouts.R (hunk 1 above)
Comment (jchojna) [reply R/layouts.R:233]: I think we could simplify it to:
`cell_widths <- rep("1fr", n_elems)`
where 1fr is a fraction unit (it takes as much place as it is left to take, so when we have e.q. five elements with 1fr width, the container gets divided on five segments with the same width

Comment (dokato) [reply R/layouts.R:233]: @przytu1 yeah, there was legacy code that I modified. thanks @jchojna , good point. Changed.

**Comment by dokato** [comment]
@przytu1 @jchojna I introduced your suggestions

**Pull request merged**